### ISSUE TITLE
Sequence contains more than one matching element

### ISSUE THREAD

**Issue by bitaxis** [esskar/Serialize.Linq#39]
I am using version 1.2.1 of Serialize.Linq.  When I try the following construct, calling ```ToExpressionNode()``` causes an ```InvalidOperationException``` to be thrown:

```c#
var feb1 = new DateTime(2015, 2, 1);
var feb15 = new DateTime(2015, 2, 15);
Expression<Func<DataPoint, bool>> expr =
    dp => dp.Timestamp >= feb1 && dp.Timestamp < feb15 && dp.AcctId == 1;
var exprNode = expr.ToExpressionNode();
```
The error message is:

```
Sequence contains more than one matching element
```

The relevant stack trace is provided below.  However, note that if I remove the ```dp.Timestamp < feb15``` condition from the expression, then calling ```ToExpressionNode()``` works just fine.  This sounds like some sort of bug?  Is there a workaround?

```
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.TryToInlineExpression(MemberExpression memberExpression, Expression& inlineExpression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.ResolveMemberExpression(MemberExpression memberExpression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.Create(Expression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode.Initialize(BinaryExpression expression)
   at Serialize.Linq.Nodes.ExpressionNode`1..ctor(INodeFactory factory, TExpression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode..ctor(INodeFactory factory, BinaryExpression expression)
   at Serialize.Linq.Factories.NodeFactory.Create(Expression expression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.Create(Expression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode.Initialize(BinaryExpression expression)
   at Serialize.Linq.Nodes.ExpressionNode`1..ctor(INodeFactory factory, TExpression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode..ctor(INodeFactory factory, BinaryExpression expression)
   at Serialize.Linq.Factories.NodeFactory.Create(Expression expression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.Create(Expression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode.Initialize(BinaryExpression expression)
   at Serialize.Linq.Nodes.ExpressionNode`1..ctor(INodeFactory factory, TExpression expression)
   at Serialize.Linq.Nodes.BinaryExpressionNode..ctor(INodeFactory factory, BinaryExpression expression)
   at Serialize.Linq.Factories.NodeFactory.Create(Expression expression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.Create(Expression expression)
   at Serialize.Linq.Nodes.LambdaExpressionNode.Initialize(LambdaExpression expression)
   at Serialize.Linq.Nodes.ExpressionNode`1..ctor(INodeFactory factory, TExpression expression)
   at Serialize.Linq.Nodes.LambdaExpressionNode..ctor(INodeFactory factory, LambdaExpression expression)
   at Serialize.Linq.Factories.NodeFactory.Create(Expression expression)
   at Serialize.Linq.Factories.TypeResolverNodeFactory.Create(Expression expression)
   at Serialize.Linq.Factories.DefaultNodeFactory.Create(Expression expression)
   at Serialize.Linq.Serializers.ExpressionConverter.Convert(Expression expression)
   at Serialize.Linq.Extensions.ExpressionExtensions.ToExpressionNode(Expression expression)
```

**Comment by bitaxis** [comment]
Funny thing is, if I just change the names of the variables, it works.  What gives???

```c#
var start = new DateTime(2015, 2, 1);
var end = new DateTime(2015, 2, 15);
Expression<Func<DataPoint, bool>> expr =
    dp => dp.Timestamp >= start && dp.Timestamp < end && dp.AcctId == 1;
var exprNode = expr.ToExpressionNode();
```

**Comment by bitaxis** [comment]
Okay.  Going back to the original snippet.  The reason it didn't work is because ```feb1``` is a substring of ```feb15```.  Even if I were to rename ```feb1``` to ```eb1```, it would still cause the exception to occur.  That is why renaming the two variables to ```start``` and ```end``` worked.

```c#
var feb1 = new DateTime(2015, 2, 1);
var feb15 = new DateTime(2015, 2, 15);
Expression<Func<DataPoint, bool>> expr =
    dp => dp.Timestamp >= feb1 && dp.Timestamp < feb15 && dp.AcctId == 1;
var exprNode = expr.ToExpressionNode();
```

Weird, eh?

**Comment by esskar** [comment]
will try to publish 1.2.2 asap

**Comment by bitaxis** [comment]
Thanks for the quick turnaround.  Looking forward to seeing 1.2.2 released.

**Comment by esskar** [comment]
I just published version 1.2.2.

**Comment by bitaxis** [comment]
Yay!
